core/src/main/scala/org/apache/spark/scheduler/DAGScheduler.scala

```diff
@@ -1885,6 +1885,14 @@ private[spark] class DAGScheduler(
               mapOutputTracker.
                 unregisterMergeResult(shuffleId, reduceId, bmAddress, Option(mapIndex))
             }
+          } else {
+            // Unregister the merge result of <shuffleId, reduceId> if
+            // the FetchFailed event contains a mapIndex of -1, and is not a
+            // MetaDataFetchException which is signified by bmAddress being null
+            if (bmAddress != null && pushBasedShuffleEnabled) {
+              mapOutputTracker.
+                unregisterMergeResult(shuffleId, reduceId, bmAddress, Option(mapIndex))
```

Comment:
Given `mapIndex` is -1, should this not be `None` passed in to `unregisterMergeResult` instead of `Option(-1)` ?
The tests are passing because you explicitly added `-1` to the bitmap, which looks incorrect.

Reply:
yep, passing in `None` now. this way we do not have to explicitly pass in -1 in the tests.